resurfemg/visualization/visualization.py

```diff
@@ -38,3 +39,77 @@ def show_my_power_spectrum(sample, sample_rate, upper_window):
     plt.plot(x_f[idx], y_f[idx])
     plt.show()
     return y_f, x_f
+
+
+def show_psd_welch(sample, sample_rate, nperseg, axis_spec):
+    """This function calculates the power spectrum density using the Welch
+    method.
+
+    :param sample: the sample array
+    :type sample: ~numpy.ndarray
+    :param sample_rate: Number of samples per second
+    :type sample_rate: int
+    :param nperseg:Length of each segment for Welch's method
+    :type nperseg: int
+    :param axis_spec: 1 for logaritmic axis, 0 for linear axis
+    :type axis_spec: int
+    :return: 'f, Pxx_den'
+    """
+    if sample.ndim != 1:
+        raise ValueError("Sample array must be 1-dimensional")
+
+    window = np.hanning(nperseg)
+    f, Pxx_den = signal.welch(sample, sample_rate,
+                              window=window, nperseg=nperseg)
+
+    if axis_spec == 1:
+        plt.semilogy(f, Pxx_den)
+        plt.ylim([0.5e-3, 1])
+        plt.xlabel('Frequency [Hz]')
+        plt.ylabel('PSD [V**2/Hz]')
+        plt.title('Power Spectral Density (Logarithmic Scale)')
+        plt.show()
+    elif axis_spec == 0:
+        plt.plot(f, Pxx_den)
+        plt.xlabel('Frequency [Hz]')
+        plt.ylabel('PSD [V**2/Hz]')
+        plt.title('Power Spectral Density (Linear Scale)')
+        plt.show()
+    else:
+        raise ValueError("Invalid axis_spec value. Please use 1"
+                         " for logarithmic axis or 0 for linear axis.")
+
+    return f, Pxx_den
```

Comment:
I am missing tests for the functions. The plotting will probably not change much over time, the signal analysis functions might. Assuring "sample", f and Pxx_den to be of the same length will assure the functions within are not broken.